controllers/etcdcluster_controller_test.go

```diff
@@ -267,8 +266,23 @@ func (s *controllerSuite) testClusterController(t *testing.T) {
 			err = try.Eventually(func() error {
 				var actual v1.PersistentVolumeClaimList
 				err := s.k8sClient.List(s.ctx, &actual, client.InNamespace(etcdCluster.Namespace))
+
+				// Since we can't control when PVCs with the pvc-protection are deleted
+				// we will remove them from this list
+				var pvcsWOProtectionFinalizer v1.PersistentVolumeClaimList
+				pvcsWOProtectionFinalizer.TypeMeta = actual.TypeMeta
+				pvcsWOProtectionFinalizer.ListMeta = actual.ListMeta
+				for _, pvc := range actual.Items {
+					// If a PVC doesn't have a deletion timestamp or it has some
+					// other finalizer we shouldn't consider it deleted
+					if pvc.DeletionTimestamp != nil && len(pvc.Finalizers) == 1 && pvc.Finalizers[0] == "kubernetes.io/pvc-protection" {
```

Comment:
I don't know all the details, but [0] looks a bit odd for me. Should a proper find solution make this expression more robust?

Reply:
Discussed offline